Feature/update

forslund opened the pull request:
- Fix Handling errors requiring a restart of the pairing process
- Add dialog specific for devices without buttons
- Add better responses if pairing is invoked while already paired
- Wait for mycroft.ready before speaking the "I'm now ready for use" dialog. This should be looked at more in the future, there's a potential problem of misleading dialog if the device is paired, the skill is reloaded and pairing is then lost and repairing is started.
- Use time.monotonic() to handle time jumps due to ntp-synchronization.

linuxipho committed:
Adds french translation

linuxipho committed:
Lets english keywords for automatic launch of the skill

forslund committed:
Update translations

forslund committed:
Switch to monotonic as time reference.

switch to time.monotonic to better handle an ntp-update during pairing.

forslund committed:
Use register intent using decorator

forslund committed:
Add already.paired dialog.

forslund committed:
Separate dialog for devices without buttons

Pairing complete message included a sentence regarding the button, for
Mark-2, picroft etc no button exists

forslund committed:
Use is_pairing from mycroft.api

forslund committed:
Handle pairing complete before skills are loaded

forslund committed:
Fix abort and restart procedure

Make sure the state variables are reset allowing a new pairing session
to start.

forslund committed:
Remove call to MycroftSkill class's shutdown

The supers' shutdown doesn't do anything anymore.

forslund commented:
Since this is a key skill it'll need a fair bit of testing.

davidwagnerkc commented:
Looks good, knocking out a few needed items with this one! Chris and I will see if we can't break this somehow and if we fail to do so I'll come back and approve. 👍

chrisveilleux commented on __init__.py:
```diff
@@ -17,13 +17,16 @@
 import mycroft.audio
 
 
+PLATFORMS_WITH_BUTTON = ['mycroft_mark_1']
```
Consider making this a tuple.  Constants should be immutable.

chrisveilleux commented on __init__.py:
```diff
@@ -17,13 +17,16 @@
 from uuid import uuid4
 from requests import HTTPError
 from adapt.intent import IntentBuilder
```
PEP8 says there should be a blank line between built-in imports and third party imports.  it also says there should be a blank line between third party imports and local package imports (Mycroft, in this case)

chrisveilleux commented on __init__.py:
```diff
@@ -43,28 +46,51 @@ def __init__(self):
+            else:
+                self.mycroft_ready = True
+
     def not_paired(self, message):
```
if a variable is not being used, consider replacing it with an underscore.  In python, this means that a variable is expected there but there is no use for it.  This could be done throughout this class.

chrisveilleux commented on dialog/en-us/wait.for.startup.dialog:
```diff
@@ -0,0 +1,2 @@
+pairing is now complete, wait one moment while skills are loading.
```
consider changing to "wait a few moments".  depending on a users' internet connection, this could take several minutes

chrisveilleux commented on __init__.py:
```diff
@@ -43,28 +46,51 @@ def __init__(self):
+
+        platform = self.config_core['enclosure'].get('platform', 'unknown')
+        if platform in PLATFORMS_WITH_BUTTON:
+            self.paired_dialog = 'pairing.paired'
```
all instance attributes should be defined in __init__().  it can be set to none or to a default value.

forslund replied:
Good catch, I attempted to do this on line 51 but between that line and this line I changed my mind of the attribute name it seems :)

forslund committed:
Minor cleanups

- Format the few existing docstrings better
- Add blanks between different kinds of imports

chrisveilleux commented on dialog/en-us/wait.for.startup.dialog:
```diff
@@ -0,0 +1,2 @@
+pairing is now complete, please wait a few moments while skills are loading.
+the device is now paired, please wait a few moment while I finish loading skills.
```
moments = plural

chrisveilleux reviewed (approved):
Just one minor comment on one of the utterances...

davidwagnerkc reviewed (approved):
Working well. Change moment -> moments to address Chris's comment and merge away.
